vike-vue/renderer/app.ts

```diff
@@ -71,7 +71,7 @@ function createVueApp(pageContext: PageContext, ssrApp = true, renderHead = fals
 
   // When doing Client Routing, we mutate pageContext (see usage of `app.changePage()` in `onRenderClient.ts`).
   // We therefore use a reactive pageContext.
-  const pageContextReactive = reactive(pageContext)
+  const pageContextReactive = reactive(pageContext as PageContext & { app: undefined })
```

Comment:
I tried using `as PageContextWithApp` here instead and indeed it doesn't work, however my intuition is that something simpler should be possible here.

Until now the pattern has been to be able to extend the `PageContext` type for everyone by doing `declare global { namespace Vike { interface PageContext { ...` and then just doing `pageContext as PageContext` at places like here. I'm not sure why this doesn't work right now. I'm running out of time today but will give another go at it tomorrow.

Reply:
I think Vue's reactivity system breaks something around the typings when passing in the entire `app`, which we're not doing anyways. But since app *might* be defined here just from looking at the typings, although we know it's not, TS still ends up with an error unless we explicitly tell it that `app` is indeed undefined.